nomos-utils/src/fisheryates.rs

```diff
@@ -0,0 +1,23 @@
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaCha20Rng;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct FisherYatesShuffle {
+    pub entropy: [u8; 32],
+}
+
+impl FisherYatesShuffle {
+    pub fn new(entropy: [u8; 32]) -> Self {
+        Self { entropy }
+    }
+
+    pub fn shuffle<T: Clone>(elements: &mut [T], entropy: [u8; 32]) {
```

Comment:
It seems that we do not need `Clone` bound here.

Reply:
Yeah, probably better to remove it then.